synapse/membrane.py

```diff
@@ -0,0 +1,151 @@
+# FIXME support ival splices
+# FIXME implement roles/groups/users/perms
+
+import copy
+import fnmatch
+import operator
+
+from synapse.eventbus import EventBus
+import synapse.lib.syntax as s_syntax
+
+class Membrane(EventBus):
+
+    def __init__(self, dst, rules, default=None):
+        '''
+        Filters Cortex splices based on rules and forwards to destination Cortex if rules allow.
+
+        Args:
+            dst: the destination Cortex to forward messages to
+            rules: the filter rules
+        '''
+        _SPLICE_NAMES = (
+            'node:add',
+            'node:del',
+            'node:prop:set',
+            'node:prop:del',
+            'node:tag:add',
+            'node:tag:del',
+            #'node:ival:set',
+            #'node:ival:del',
+        )
+        _DEFAULT_RULE = [{'query': ''}]
+        _DEFAULT_RULES = {'': {k: copy.deepcopy(_DEFAULT_RULE) for k in _SPLICE_NAMES}}
+
+        EventBus.__init__(self)
+        self.dst = dst
+
+        self.rules = copy.deepcopy(_DEFAULT_RULES)
+        if rules:
+            for key in rules:
+                for rule in rules[key]:
+                    if self._is_valid_rule(rule):
+                        self.rules[''][key].append(rule)
+
+        self.opers = {
```

Comment:
We probably want these to work exactly like the opers in storm. This stuff seems to be scattered across a few files, should it be its own cmpr.py?